### PR TITLE
Oy2 18553 - Qualifier Refactor and Year movement

### PR DESCRIPTION
**Pull request by daniel-belcher**
## Purpose

Tweak the qualifiers to use a data driven approach for rendering to lower amount of pages/views for the non-unique aspects. Also adds the Qualifiers to both 2021 and 2022 as individual file sets to allow for changes between years.

#### Linked Issues to Close

[ticket to close](https://qmacbis.atlassian.net/browse/OY2-18553?atlOrigin=eyJpIjoiMTE5NjFiZjJhMGIxNDIzMGFiN2MxMjhiYWJiY2I3ZGMiLCJwIjoiaiJ9)

## Approach

Rework of the Delivery Systems component to be generic and allow for dynamic number of fields between the qualifiers. Change to routes to dynamically make the routes for qualifiers so differing years can have changes in the components. Qualifiers are now treated as measures in front end display to match back-end treatment for ease of data assignment and interaction.

## Assorted Notes/Considerations

Need to be careful when abstracting older view paths to be dynamic per year with the routing table. Reuse of the MeasureWrapper or similar abstractions of form logic help significantly with repurposing/reusability of the components.

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/hooks/api/useGetMeasures.tsx [hunk 1]
```diff
@@ -24,6 +24,7 @@ export const useGetMeasures = () => {
     coreSet: coreSetPath,
   } = usePathParams();
   const { state, year, coreSetId } = useParams();
+  console.log(year || yearPath);
```
Comment (benjaminpaige) [inline services/ui-src/src/hooks/api/useGetMeasures.tsx:27]: can prob remove

File: services/ui-src/src/measures/2021/Qualifiers/Common/audit.tsx [hunk 2]
```diff
@@ -25,13 +27,15 @@ export const CloseButton = ({ onClick }: { onClick: () => void }) => (
...
+  year: string;
 }
 
-export const Audit = ({ type }: Props) => {
+export const Audit = ({ type, year }: Props) => {
   const { fields, append, remove, replace } = useFieldArray({
     name: "CoreSetMeasuresAuditedOrValidatedDetails",
   });
   const { data, isLoading } = useGetMeasures();
+  console.group("measureList");
```
Comment (benjaminpaige) [inline services/ui-src/src/measures/2021/Qualifiers/Common/audit.tsx:38]: ooooh console.group fancy

File: services/ui-src/src/measures/2021/Qualifiers/data.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+  },
+};
+
+export const Data = {
+  HHCS: HomeData,
+  ACS: AdultData,
+  CCS: ChildData,
+  CCSC: ChildChipData,
+  CCSM: ChildMedicaidData,
+};
```
Comment (benjaminpaige) [inline services/ui-src/src/measures/2021/Qualifiers/data.ts:191]: love this file

**Review by benjaminpaige — COMMENTED** [review]
couple of comments. Theres a lot going on here. If you could add just a couple comments to the code added in the routes and measurewrapper that would prob be super helpful.

**Review by abelzer212 — APPROVED** [review]
Could you take a look at the failing qualifier tests while this is in manual testing?

**Review by abelzer212 — APPROVED** [review]

**Pull request merged**